python-package/setup.py

```diff
@@ -63,33 +76,56 @@ def compile_cpp(use_mingw=False, use_gpu=False):
     os.makedirs("build_cpp")
     os.chdir("build_cpp")
 
-    cmake_cmd = "cmake "
-    build_cmd = "make _lightgbm"
+    logger.info("Starting to compile the library.")
+
+    cmake_cmd = ["cmake", "../lightgbm/"]
     if use_gpu:
-        cmake_cmd += " -DUSE_GPU=ON "
+        cmake_cmd.append("-DUSE_GPU=ON")
     if os.name == "nt":
         if use_mingw:
-            cmake_cmd += " -G \"MinGW Makefiles\" "
-            os.system(cmake_cmd + " ../lightgbm/")
-            build_cmd = "mingw32-make.exe _lightgbm"
+            logger.info("Starting to compile with CMake and MinGW.")
+            status = silent_call(cmake_cmd + ["-G", "MinGW Makefiles"])
```

Comment:
the name `status` is a little confusing, can it be `succeed` or `fail` or sth like this? (I don't read through all codes because I am not sure about those VS install logic).

Reply:
Will `exitcode` be OK?

Reply:
ignore this.